src/Shared.CLI/Helpers/RepoSearch.cs

```diff
@@ -28,16 +25,16 @@ namespace Microsoft.CST.OpenSource.Shared
     /// </summary>
     public class RepoSearch
     {
-        public RepoSearch(IHttpClientFactory httpClientFactory)
-        {
-            HttpClientFactory = httpClientFactory;
-        }
-
-        public RepoSearch() : this(new DefaultHttpClientFactory())
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RepoSearch"/> class.
+        /// </summary>
+        /// <param name="projectManagerFactory">The <see cref="ProjectManagerFactory"/> to generate the project managers with.</param>
+        public RepoSearch(ProjectManagerFactory? projectManagerFactory = null)
         {
+            _projectManagerFactory = projectManagerFactory ?? new ProjectManagerFactory();
```

Comment:
This is the pattern we've generally adopted when provided a null client/'manager.

Reply:
This was in response to https://github.com/microsoft/OSSGadget/pull/326#discussion_r844243432 right? I don't think I have anything to change here but I could be reading what you commented wrong.

Reply:
This was just extra context for another comment. No actoin needed on this line.